tunnel/intercept/interceptor.go

```diff
@@ -90,86 +88,21 @@ func addrBits(ip net.IP) int {
 	return 0
 }
 
-func NewInterceptAddress(service *entities.Service, protocol string, resolver dns.Resolver) (*interceptAddress, error) {
-	if service.ClientConfig == nil {
-		return nil, errors.Errorf("no client configuration for service %v", service.Name)
-	}
-
-	if service.ClientConfig.Hostname == "" {
-		return nil, errors.Errorf("client configuration missing hostname for service %v", service.Name)
-	}
-
-	if service.ClientConfig.Port < 1 || service.ClientConfig.Port > math.MaxUint16 {
-		return nil, errors.Errorf("client configuration has invalid port %v for service %v", service.ClientConfig.Port, service.Name)
-	}
-
-	ip, err := getInterceptIP(service.ClientConfig.Hostname, resolver)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get intercept IP address: %v", err)
-	}
-
-	prefixLen := addrBits(ip)
-	ipNet := net.IPNet{IP: ip, Mask: net.CIDRMask(prefixLen, prefixLen)}
-	addr := interceptAddress{cidr: ipNet.String(), port: service.ClientConfig.Port, protocol: protocol}
-	return &addr, nil
-}
-
-type LUT map[interceptAddress]InterceptedService
-
-func NewLUT() LUT {
-	return make(map[interceptAddress]InterceptedService)
-}
-
-// Return the service that is being intercepted at the given address, if any.
-func (m LUT) GetByAddress(addr net.Addr) (*InterceptedService, error) {
-	//var proto string
-	var ip net.IP
-	var port int
-	var protocol string
-	switch addrType := addr.(type) {
-	case *net.TCPAddr:
-		ip = addrType.IP
-		port = addrType.Port
-		protocol = "tcp"
-	case *net.UDPAddr:
-		ip = addrType.IP
-		port = addrType.Port
-		protocol = "udp"
-	default:
-		return nil, fmt.Errorf("unsupported address type: %v", addrType)
-	}
-
-	prefixLen := addrBits(ip)
-	ipNet := &net.IPNet{IP: ip, Mask: net.CIDRMask(prefixLen, prefixLen)}
-	cidr := ipNet.String()
-
-	s, ok := m[interceptAddress{cidr, port, protocol}]
-	if !ok {
-		return nil, fmt.Errorf("address %s is not being intercepted", addr.String())
-	}
-	return &s, nil
-}
-
-func (m LUT) GetByName(serviceName string) []InterceptedService {
-	var r []InterceptedService
-	for _, service := range m {
-		if serviceName == service.Name {
-			r = append(r, service)
+func GetInterceptAddresses(service *entities.Service, protocol string, resolver dns.Resolver) ([]*InterceptAddress, error) {
```

Comment:
Should protocol be populated with intercept.v1 details?
What about client.v1 configuration?